apps/antalmanac/src/stores/Schedules.ts

```diff
@@ -461,20 +461,23 @@ export class Schedules {
 
             // Get the course info for each course
             const courseInfoDict = new Map<string, { [sectionCode: string]: CourseInfo }>();
-            for (const [term, courseSet] of Object.entries(courseDict)) {
+
+            const websocRequests = Object.entries(courseDict).map(async ([term, courseSet]) => {
                 const sectionCodes = Array.from(courseSet);
```

Comment:
If this variable name is `sectionCodes`, then I think it should correspond directly with the parameter passed to `queryWebsoc`. Suggestion is continued below.

```suggestion
                const sectionCodes = Array.from(courseSet).join(',');
```